libclc/CMakeLists.txt

```diff
@@ -52,170 +115,51 @@ set( LIBCLC_TARGETS_ALL
   nvptx64--
   nvptx--nvidiacl
   nvptx64--nvidiacl
-  spirv-mesa3d-
-  spirv64-mesa3d-
 )
 set( LIBCLC_TEST_TARGETS_ALL
   nvptx--nvidiacl
   nvptx64--nvidiacl
   amdgcn--amdhsa
 )
 
-set( LIBCLC_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR} )
-set( LIBCLC_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR} )
-
-# $&$ maybe delete
 # mesa3d environment is only available since LLVM 4.0
 if( LLVM_PACKAGE_VERSION VERSION_GREATER_EQUAL 4.0.0 )
   list( APPEND LIBCLC_TARGETS_ALL amdgcn-mesa-mesa3d )
 endif()
-# $&$ end delete
 
-set( LIBCLC_TARGETS_TO_BUILD "all"
-    CACHE STRING "Semicolon-separated list of targets to build, or 'all'." )
-
-option( ENABLE_RUNTIME_SUBNORMAL "Enable runtime linking of subnormal support."
-OFF )
+# spirv-mesa3d and spirv64-mesa3d targets can only be built with the (optional)
```

Comment:
above we throw a fatal error if we can't find these tools. Are we sure they're optional?

Reply:
Upstream the tool definitely is optional. For us it's not optional if you use the `LIBCLC_CUSTOM_LLVM_TOOLS_BINARY_DIR` option which upstream doesn't have.

So we handle the requirements on that tool slightly differently, but again we've been deleting or not integrating upstream code (handling these targets and others like [clspv/clspv64](https://github.com/llvm/llvm-project/blob/main/libclc/CMakeLists.txt#L22-L23) - [for example](https://github.com/llvm/llvm-project/blob/main/libclc/CMakeLists.txt#L254-L256)) that we'll need to be able to accommodate if we want to merge our changes back to LLVM.

Reply:
ack